utils/docker-compose/README.md

```diff
@@ -0,0 +1,228 @@
+```
+If you want to build the peer image using your local copy of your repo you can
+use the same build args, but specify `file:///tmp/build-src/.git` as the
+`FPC_REPO_URL`. You will also need to create the image at the root of this repo
+so that the local repo will be in the build context for the docker daemon.
+```
+cd $FPC_PATH
+docker build -t hyperledger/fabric-private-chaincode-dev --build-arg FPC_REPO_URL=file:///tmp/build-src/.git --build-arg FPC_REPO_BRANCH=$(git rev-parse --abbrev-ref HEAD) .
+docker build -t hyperledger/fabric-peer-fpc -f utils/docker/peer/Dockerfile .
+```
+2. Generate the cryptographic material needed for the network by running the
+[generate](scripts/generate.sh) script. Cryptogen will be used to generate all the
+credentials needed based on the configuration filesabove and place them in the
+`network-config/crypto-config` directory.  Configtxgen will be used to create
+the genesis block which is used to start up the orderer as well as the peer
+create channel configuration transaction. These will be placed in the
+`network-config/config` directory. The `crypto-config` & `config` directory will
+be mounted into every container of the FPC network as specified in the
+docker-compose file. **This script is not
+idempotent and will delete the contents of `crypto-config` & `config` when run
+to ensure a clean start.**
+```
+cd $FPC_PATH/utils/docker-compose
+scripts/generate.sh
+```
+3. Start the network. Run the [start](scripts/start.sh) script. This will use
+docker-compose to start the network as well as starting the channel `mychannel`.
+By default, this script will use FPC peers. If non FPC peers are desired, set
+`$USE_FPC` to `false`.
+```
+.scripts/start.sh
+```
+
+## Deploying your FPC Chaincode
+The examples directory has been mounted into the peer container for convenience,
+under `/opt/examples`. All of these steps should be done within the peer container.
+
+1. Exec into the peer container.
+```
+docker exec -it peer0.org1.example.com bash
+```
+
+2. Follow the [steps](../../examples/README.md) in the tutorial to build your
+chaincode within the peer container. Do not continue to the testing step.
+
+3. Set up the peer command environment variable for convenience. `$FPC_CMDS` is
+already defined in the container.
+```
+export PEER_CMD=$FPC_CMDS/peer.sh
+```
+
+4. Set environment variable to use the admin credentials and set the
+orderer address.
+```
+export CORE_PEER_MSPCONFIGPATH=/etc/hyperledger/msp/users/Admin@org1.example.com/msp
+export ORDERER_ADDR=orderer.example.com:7050
+```
+
+5. Install your chaincode
+```
+${PEER_CMD} chaincode install -l fpc-c -n helloworld_test -v 0 -p examples/helloworld/_build/lib
+```
+
+6. Instantiate your chaincode
+```
+${PEER_CMD} chaincode instantiate -o orderer.example.com:7050 -C mychannel -n helloworld_test -v 0 -c '{"Args":["init"]}' -V ecc-vscc
```

Comment:
you might want to use ${ORDERER_ADDR} here?

Reply:
Thanks for catching that! I will remove the flag since it is already defined as an env var, we don't need to specify it will the flag.